### ISSUE TITLE
TypeError: unhashable type: 'dict' when patching dict

### ISSUE THREAD

**Issue by rooterkyberian** [Simplistix/testfixtures#101]
The same code worked fine with 5.4.0, now is broken on 6.4.0 :
```
import unittest
from unittest.mock import patch

from testfixtures import tempdir


class Test(unittest.TestCase):
    @patch('sre_parse.FLAGS', {
        'x': 1,
    })
    @tempdir()
    def test_my(self, output_dir):
        pass
```

results in 

```
Error
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib64/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File ".../lib/python3.6/site-packages/testfixtures/utils.py", line 74, in patched
    elif patching.new in defaults and added < to_add:
TypeError: unhashable type: 'dict'
```

**Comment by cjw296** [comment]
Is this in a virtualenv? It, would be useful to know exactly what packages you have installed and if you are, a `pip freeze` would give that.

I can see you're using Python 3.6, but can you let me know the specific version and platform you're on?

There's quite a few releases between 5.4.0 and 6.4.0. Please can you isolate the exact version where this problem first showed up? I suspect it's 6.0.0, but would be good to confirm.

As an observation, `create=True` is unnecessary as it's the default.

**Comment by cjw296** [comment]
This is fixed in 6.4.1, which will be on PyPI as soon as https://travis-ci.org/Simplistix/testfixtures/builds/471876836 completes.

**Comment by rooterkyberian** [comment]
It works great.

That was very fast - thank you for your hard work!
